pkg/storage/ingest/pusher.go

```diff
@@ -30,10 +29,11 @@ type Pusher interface {
 
 type parsedRecord struct {
```

Comment:
`parsedRecord` is only only used in `pusherConsumer.Consume`. You can move it into that method if you want to reduce its scope. Not sure if you like that kind of type definitions

Reply:
I moved it.

Reply:
Done in https://github.com/grafana/mimir/pull/9020/commits/9c310be3c4388cb3838034dfcf76547087a73359